opensrp-chw-core/src/main/java/org/smartregister/chw/core/presenter/CoreFamilyPlanningProfilePresenter.java

```diff
@@ -0,0 +1,63 @@
+package org.smartregister.chw.core.presenter;
+
+import org.smartregister.chw.core.contract.CoreFamilyPlanningMemberProfileContract;
+import org.smartregister.chw.core.utils.CoreConstants;
+import org.smartregister.chw.fp.contract.BaseFpProfileContract;
+import org.smartregister.chw.fp.domain.FpMemberObject;
+import org.smartregister.chw.fp.presenter.BaseFpProfilePresenter;
+import org.smartregister.family.util.Utils;
+import org.smartregister.repository.AllSharedPreferences;
+import org.smartregister.util.FormUtils;
+
+import java.lang.ref.WeakReference;
+
+import timber.log.Timber;
+
+public class CoreFamilyPlanningProfilePresenter extends BaseFpProfilePresenter implements CoreFamilyPlanningMemberProfileContract.Presenter {
+    private BaseFpProfileContract.Interactor interactor;
+    private WeakReference<BaseFpProfileContract.View> view;
+    private FormUtils formUtils;
+    private FpMemberObject fpMemberObject;
+
+    public CoreFamilyPlanningProfilePresenter(BaseFpProfileContract.View view, BaseFpProfileContract.Interactor interactor, FpMemberObject fpMemberObject) {
+        super(view, interactor, fpMemberObject);
+        this.interactor = interactor;
+        this.view = new WeakReference<>(view);
+        this.fpMemberObject = fpMemberObject;
+    }
+
+    @Override
+    public void createReferralEvent(AllSharedPreferences allSharedPreferences, String jsonString) throws Exception {
+        ((CoreFamilyPlanningMemberProfileContract.Interactor) interactor).createReferralEvent(allSharedPreferences, jsonString, fpMemberObject.getBaseEntityId());
+    }
+
+    @Override
```

Comment:
This method could return a null, so i suggest annotate it with @Nullable

Reply:
The rest seems fine to me

Reply:
Which method specifically?

Reply:
getView

Reply:
if (view != null) {
            return (CoreFamilyPlanningMemberProfileContract.View) view.get();
        } else {
            return null;
        }